src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c

```diff
@@ -30,17 +29,92 @@ const ActorInit Bg_Ingate_InitVars = {
     (ActorFunc)BgIngate_Update,
     (ActorFunc)BgIngate_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ingate/BgIngate_SetupAction.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ingate/BgIngate_Init.s")
+extern UNK_TYPE D_060011B8;
+extern Gfx D_06001040[];
+
+void BgIngate_SetupAction(BgIngate* this, BgIngateActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void BgIngate_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgIngate* this = THIS;
+
+    s32 pad;
+    s32 sp32 = 0;
+
+    DynaPolyInfo_SetActorMove(&this->dyna, 0);
+    DynaPolyInfo_Alloc(&D_060011B8, &sp32);
+
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, sp32);
+
+    if ((globalCtx->sceneNum != SACRED_FOREST_MEADOW_AFTER_FOREST_BLUE_WARP || LINK_IS_CHILD) ||
```

Comment:
you grabbed this from a cutscene enum, this is just a regular scene. you can find these in z64scene.h
this one is SCENE_SPOT20

Reply:
Done

Reply:
Thanks for pointing that out